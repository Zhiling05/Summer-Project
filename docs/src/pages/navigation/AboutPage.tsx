// src/pages/navigation/SettingsPage.tsx
export default function AboutPage() {
  return <h1>SettingsPage</h1>;
}
