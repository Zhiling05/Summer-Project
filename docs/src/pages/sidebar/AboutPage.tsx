// src/pages/sidebar/AboutPage.tsx
import Header from '../../components/Header';
import BackButton from '../../components/BackButton';


export default function AboutPage() {
  return (
    <div className="about-page">
      <Header title="About Us" />
            <BackButton />
      <h1 className="about-title">About Us Page</h1>
      
      <section className="about-section">
        <h2 className="about-heading">DIPP Research Team</h2>
        <p className="about-description">
          The DIPP is led by a team of passionate researchers dedicated to improving healthcare.
        </p>
        
        <h3 className="team-title">Team members:</h3>
        <ul className="team-list">
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
        </ul>
      </section>
    </div>
  );
}
