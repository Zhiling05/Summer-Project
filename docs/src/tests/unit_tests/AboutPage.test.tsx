import { render, screen } from '@testing-library/react';
import AboutPage from '../../pages/navigation/AboutUsPage';

describe('AboutPage', () => {
    it('renders without crashing', () => {
        render(<AboutPage />);
        // throws error
    });

    it('renders the correct heading', () => {
        render(<AboutPage />);
        const heading = screen.getByRole('heading', { name: /SettingsPage/i });
        expect(heading).toBeInTheDocument();
    });

    it('does not render unexpected elements', () => {
        render(<AboutPage />);
        // no other headings...
        expect(screen.queryByRole('heading', { name: /About/i })).not.toBeInTheDocument();
        expect(screen.queryByRole('button')).not.toBeInTheDocument();
        expect(screen.queryByRole('link')).not.toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<AboutPage />);
        expect(asFragment()).toMatchSnapshot();
    });
});
