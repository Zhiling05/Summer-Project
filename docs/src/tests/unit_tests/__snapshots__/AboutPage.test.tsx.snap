// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    About us page
  </h1>
</DocumentFragment>
`;
